src/components/History/History.vue

```diff
@@ -0,0 +1,551 @@
+        :class="{ 'move-image': secondPosition !== 0 }"
+        class="history-image-element"
+      >
+        <img
+          v-if="secondImage"
+          class="history-image display-image"
+          :class="getOpacityFullSecondPosition"
+          @click="comeBackTo(secondImage)"
+          :src="secondImage.data"
+        />
+      </div>
+      <div
+        style="position: absolute; left: 135px"
+        :style="{ transform: 'translateX(' + -(45 * thirdPosition) + 'px)' }"
+        :class="{ 'move-image': thirdPosition !== 0 }"
+        class="history-image-element"
+      >
+        <img
+          v-if="thirdImage"
+          class="history-image display-image"
+          :class="getOpacityFullThirdPosition"
+          @click="comeBackTo(thirdImage)"
+          :src="thirdImage.data"
+        />
+      </div>
+      <div
+        style="position: absolute; left: 135px"
+        :style="{ transform: 'translateX(' + -(45 * forthPosition) + 'px)' }"
+        :class="{ 'move-image': forthPosition !== 0 }"
+        class="history-image-element"
+      >
+        <img
+          v-if="forthImage"
+          class="history-image display-image"
+          :class="getOpacityFullForthPosition"
+          @click="comeBackTo(forthImage)"
+          :src="forthImage.data"
+        />
+      </div>
+    </div>
+  </div>
+
+  <div v-if="displayAllHistory" class="history-layout" :style="fullHistory">
+    <div class="left-arrow" @click="leftMove()" />
+    <div class="full-history-display">
+      <div v-for="(value, index) in currentHistory" :key="index">
+        <div
+          style="position: absolute; width: 64; height: 81px"
+          :style="{
+            left: 68 * index + 'px',
+            transform: 'translateX(' + -(68 * position) + 'px)',
+          }"
+        >
+          <img
+            class="history-image-full"
+            style="height: 69px; width: 60px;"
+            @click="comeBackTo(value)"
+            :src="value.data"
+          />
+        </div>
+      </div>
+    </div>
+    <div class="right-arrow" @click="rightMove()" />
+  </div>
+</template>
+
+<script>
+import { mapGetters, mapState } from "vuex";
+export default {
+  name: "History",
+  props: {
+    topPosition: Number,
+    leftPosition: Number,
+    fullWidth: Number,
+    displayAllHistory: Boolean,
+  },
+  data() {
+    return {
+      firstImage: undefined,
+      firstPosition: 0,
+      secondImage: undefined,
+      secondPosition: 0,
+      thirdImage: undefined,
+      thirdPosition: 0,
+      forthImage: undefined,
+      forthPosition: 0,
+      currentHistoryLength: 0,
+      position: 0,
+    };
+  },
+  watch: {
+    history: {
+      handler(historyElement) {
+        if (historyElement.length > 3) {
+          if (this.currentHistoryLength !== historyElement.length) {
+            if (this.currentHistory.length % 4 === 1) {
+              setTimeout(() => (this.secondImage = undefined), 1000);
```

Comment:
why setTimeout?

Reply:
The timeout is used in this method for animation purpose. The image appears or disappears only when the left move animation is done.